core/node/da_clients/src/eigen/sdk.rs

```diff
@@ -1,76 +1,150 @@
     disperser::{
+        self,
         authenticated_request::Payload::{AuthenticationData, DisperseRequest},
         disperser_client::DisperserClient,
-        AuthenticatedReply, BlobAuthHeader, BlobVerificationProof, DisperseBlobReply,
+        AuthenticatedReply, BlobAuthHeader,
     },
+    verifier::VerificationError,
 };
 
 #[derive(Debug, Clone)]
-pub struct RawEigenClient {
+pub(crate) struct RawEigenClient {
     client: DisperserClient<Channel>,
-    polling_interval: Duration,
     private_key: SecretKey,
-    account_id: String,
+    pub config: EigenConfig,
+    verifier: Verifier,
+    get_blob_data: Arc<dyn GetBlobData>,
 }
 
 pub(crate) const DATA_CHUNK_SIZE: usize = 32;
 
 impl RawEigenClient {
-    pub(crate) const BUFFER_SIZE: usize = 1000;
+    const BLOB_SIZE_LIMIT: usize = 1024 * 1024 * 2; // 2 MB
 
     pub async fn new(
-        rpc_node_url: String,
-        inclusion_polling_interval_ms: u64,
         private_key: SecretKey,
+        config: EigenConfig,
+        get_blob_data: Arc<dyn GetBlobData>,
     ) -> anyhow::Result<Self> {
         let endpoint =
-            Endpoint::from_str(rpc_node_url.as_str())?.tls_config(ClientTlsConfig::new())?;
-        let client = DisperserClient::connect(endpoint)
-            .await
-            .map_err(|e| anyhow::anyhow!("Failed to connect to Disperser server: {}", e))?;
-        let polling_interval = Duration::from_millis(inclusion_polling_interval_ms);
+            Endpoint::from_str(config.disperser_rpc.as_str())?.tls_config(ClientTlsConfig::new())?;
+        let client = DisperserClient::connect(endpoint).await?;
+
+        let verifier_config = VerifierConfig {
+            rpc_url: config
+                .eigenda_eth_rpc
+                .clone()
+                .ok_or(anyhow::anyhow!("EigenDA ETH RPC not set"))?,
+            svc_manager_addr: config.eigenda_svc_manager_address,
+            max_blob_size: Self::BLOB_SIZE_LIMIT as u32,
+            g1_url: Url::parse(&config.g1_url)?,
+            g2_url: Url::parse(&config.g2_url)?,
+            settlement_layer_confirmation_depth: config.settlement_layer_confirmation_depth,
+            private_key: hex::encode(private_key.secret_bytes()),
+            chain_id: config.chain_id,
+        };
 
-        let account_id = get_account_id(&private_key);
+        let url = SensitiveUrl::from_str(&verifier_config.rpc_url)?;
+        let query_client: Client<L1> = Client::http(url)?.build();
+        let query_client = Box::new(query_client) as Box<DynClient<L1>>;
+        let signing_client = PKSigningClient::new_raw(
+            K256PrivateKey::from_bytes(zksync_types::H256::from_str(
+                &verifier_config.private_key,
+            )?)?,
+            verifier_config.svc_manager_addr,
+            Verifier::DEFAULT_PRIORITY_FEE_PER_GAS,
+            SLChainId(verifier_config.chain_id),
+            query_client,
+        );
 
+        let verifier = Verifier::new(verifier_config, Arc::new(signing_client))
```

Comment:
From the dependency injection perspective, it would be optimal to pass `Arc<dyn VerifierClient>` as an argument to the constructor.

Reply:
Fixed here: https://github.com/lambdaclass/zksync-era/pull/386